app/models/bulk_assignment_creation.rb

```diff
@@ -83,7 +93,10 @@ def identifier_type
 
   def existing_assignments
     ensure_identifiers
-    Assignment.where(visitor_id: existing_visitor_ids, split: split).where.not(variant: variant)
+    Assignment.where(visitor_id: existing_visitor_ids, split: split).where.not(
+      Assignment.arel_table[:variant].eq(variant)
+      .and(Assignment.arel_table[:force].eq(force))
+    )
```

Comment:
this change is the only thing i can see in here that doesn't appear to be tested. worth adding a test that a preexisting "force" assignment doesn't change after a bulk assignment? (i think that's what this is intending)

Reply:
oh no. i see that this is saying that if the inputs are "variant: yes, force: true" exclude existing assignments that match that exactly. so, if there's an existing assignment for "variant: yes, force: false" then this will still scoop that one up. coolcoolcool.